tests/quic/testExample.nim

```diff
@@ -8,23 +8,30 @@ import ../helpers/certificate
 suite "examples from Readme":
   test "outgoing and incoming connections":
     proc outgoing() {.async.} =
-      let connection = await dial(initTAddress("127.0.0.1:12345"))
+      let cb = proc(derCertificates: seq[seq[byte]]): bool {.gcsafe.} =
+        # TODO: implement custom certificate validation
+        return derCertificates.len > 0
+
+      let customCertVerif: CertificateVerifier = CustomCertificateVerifier.init(cb)
+
+      let tlsConfig = TLSConfig.init(certificateVerifier = Opt.some(customCertVerif))
+      let connection = await dial(initTAddress("127.0.0.1:12345"), tlsConfig)
```

Comment:
what do you think about following usage:

```
let tlsCfg = TLSConfig.init(certificateVerifier = Opt.some(customCertVerif))
...
let quicClient = newClient(tlsCfg)
let connection = quicClient.dial(address)
...
let stream = await connection.openStream()
````

Reply:
and server would look like:

```
let quicServer = newServer(tlsCfg)
let listener = quicServer.listen(address)

for ... 
  let connection = await listener.accept()
```

Reply:
I followed the spirit of this suggestion with some changes in https://github.com/vacp2p/nim-quic/pull/56/commits/c3e770959ed02a6d53cf978db4485f7dcce1a473